packages/3d-web-client-core/src/character/CharacterSockets.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  ModelLoader,
+  type MMLCharacterDescription,
+  type MMLCharacterDescriptionPart,
+} from "@mml-io/3d-web-avatar";
+import { Bone, Euler, Object3D, Vector3, MathUtils } from "three";
+
+export class CharacterSockets {
+  private modelLoader: ModelLoader = new ModelLoader();
+  private attachments: Map<string, Object3D> = new Map();
+  private availableBones: Map<string, Bone> = new Map();
+
+  constructor(
+    private mesh: Object3D,
+    private characterDescription: Partial<MMLCharacterDescription> | null,
+  ) {
+    this.mesh.traverse((child) => {
+      const asBone = child as Bone;
+      if (asBone.isBone) {
+        this.availableBones.set(child.name, asBone);
+      }
+    });
+
+    if (this.characterDescription?.parts) {
+      this.setAttachments(this.characterDescription.parts);
+    }
+  }
+
+  private async setAttachments(parts: MMLCharacterDescriptionPart[]): Promise<void> {
+    parts.forEach(async (part) => {
+      if (part.socket?.socket && this.availableBones.has(part.socket.socket)) {
```

Comment:
I think this should handle the case where there isn't a valid bone by adding it to the root (which is effectively between the feet)

Reply:
@MarcusLongmuir Nice catch! Fixed (I'm `console.warning` it though).